src/components/DappBrowser/control-panel/ControlPanel.tsx

```diff
@@ -429,13 +426,12 @@ const HomePanel = memo(function HomePanel({
     // Check if it's different to the globally selected wallet
     if (selectedWallet.uniqueId !== accountAddress) {
       // switch to selected wallet
-      const p1 = dispatch(walletsSetSelected(walletInPanel));
-      const p2 = dispatch(addressSetSelected(selectedWallet.uniqueId));
-      await Promise.all([p1, p2]);
+      setSelectedWallet(walletInPanel);
```

Comment:
could this be consolidated with the extra optional `address` param you expose on `setSelectedWallet` as in: `setSelectedWallet(walletInPanel, selectedWallet.uniqueId)`?